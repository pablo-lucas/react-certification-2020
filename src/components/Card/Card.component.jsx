import React from 'react';
import { useHistory } from 'react-router';
import { Card as SemanticCard, Image } from 'semantic-ui-react';

const Card = ({ video, onVideoSelect, isFromFavorite }) => {
  const history = useHistory();

  const onLinkTo = () => {
    const linkTo = isFromFavorite
      ? `/favorites/${video.id.videoId}`
      : `/watch/${video.id.videoId}`;

    history.push(linkTo);
  };

  return (
    <SemanticCard
      fluid
      as="div"
      style={{ cursor: 'pointer' }}
      onClick={() => {
        onVideoSelect(video);
        onLinkTo();
      }}
    >
      <Image src={video.snippet.thumbnails.medium.url} wrapped ui={false} />
      <SemanticCard.Content>
        <SemanticCard.Header>
          <h4>{video.snippet.title}</h4>
        </SemanticCard.Header>
        <SemanticCard.Meta>
          <span className="date">{video.snippet.channelTitle}</span>
        </SemanticCard.Meta>
      </SemanticCard.Content>
    </SemanticCard>
  );
};

export default Card;
