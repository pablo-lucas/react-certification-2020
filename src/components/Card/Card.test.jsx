import React from 'react';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Card from './Card.component';
import { video } from '../../__mocks__/videoMock';

describe('Card component', () => {
  it('Matches the snapshot', () => {
    const card = create(<Card video={video} isFromFavorite={false} />);
    expect(card.toJSON()).toMatchSnapshot();
  });

  it('renders provided properties', () => {
    render(<Card video={video} isFromFavorite={false} />);

    expect(screen.getByText(video.snippet.title)).toBeTruthy();
    expect(screen.getByRole('img').getAttribute('src')).toBe(
      video.snippet.thumbnails.medium.url
    );
  });
});
