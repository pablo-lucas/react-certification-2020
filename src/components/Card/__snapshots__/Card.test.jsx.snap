// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Matches the snapshot 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <div
    className="image"
  >
    <img
      src="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <h4>
        Video Tour | Welcome to Wizeline Guadalajara
      </h4>
    </div>
    <div
      className="meta"
    >
      <span
        className="date"
      >
        Wizeline
      </span>
    </div>
  </div>
</div>
`;
